### PR TITLE
feat: Integrate Survicate

### PR DESCRIPTION
**Pull request by matthewmuscat**
## Description
Integrate Survicate into our documentation site to deliver a Documentation Quality Score unit of measure.

This PR adds:
- Rating widget to top and bottom of our docs
- Integrates with Survicate
- Supports deploy contexts for environment variables via Netlify

## Resolved issues
Jira: https://immutable.atlassian.net/browse/DX-1046

### Before submitting the PR, please consider the following:
- [x] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [ ] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [ ] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

**Review by ckailash — APPROVED** [review]
LGTM :shipit:

File: src/theme/DocBreadcrumbs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        {children}
+      </span>
+    );
+  }
+  return href ? (
+    <Link className={className} href={href} itemProp="item">
+      <span itemProp="name">{children}</span>
+    </Link>
+  ) : (
+    // TODO Google search console doesn't like breadcrumb items without href.
```
Comment (ckailash) [inline src/theme/DocBreadcrumbs/index.tsx:32]: What is the plan with these TODOs?

Comment (matthewmuscat) [reply src/theme/DocBreadcrumbs/index.tsx:32]: These are included as part of the [Swizzle'd](https://docusaurus.io/docs/swizzling) component from Docusaurus. These can be ignored.

**Review by zatlite — APPROVED** [review]

File: docusaurus.config.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // @ts-check
 // Note: type annotations allow type checking and IDEs autocompletion
+require('dotenv').config();
```
Comment (CodeSchwert) [inline docusaurus.config.js:3]: Can we use the [docusaurus dotenv plugin](https://www.npmjs.com/package/docusaurus-plugin-dotenv) instead?

Comment (matthewmuscat) [reply docusaurus.config.js:3]: See [this](https://stackoverflow.com/a/70356791/8023887), it's not as supported/popular, I would rather just use `dotenv`.

**Review by CodeSchwert — APPROVED** [review]

**Pull request merged**